homeassistant/components/bryant_evolution/config_flow.py

```diff
@@ -0,0 +1,56 @@
+"""Config flow for Bryant Evolution integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from evolutionhttp import BryantEvolutionLocalClient
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigFlowResult
+from homeassistant.const import CONF_FILENAME
+
+from .const import CONF_SYSTEM_ID, CONF_ZONE_ID, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_FILENAME, default="/dev/ttyUSB0"): str,
+        vol.Required(CONF_SYSTEM_ID, default=1): vol.All(int, vol.Range(min=1)),
+        vol.Required(CONF_ZONE_ID, default=1): vol.All(int, vol.Range(min=1)),
+    }
+)
+
+
+async def _can_reach_device(filename: str) -> bool:
+    """Return whether we can reach the device at the given filename."""
+    # Verify that we can read S1Z1 to check that the device is valid.
+    try:
+        client = await BryantEvolutionLocalClient.get_client(1, 1, filename)
+        return await client.read_hvac_mode() is not None
+    except FileNotFoundError:
+        _LOGGER.error("Could not open %s: not found", filename)
+        return False
```

Comment:
So wait, why do we request zone and system id if we can already access it with `1 ,1`? What is the context of those values?

Reply:
The system and zone id specify which specific climate-like thing we are accessing through the bridge; there can be up to 16 ((1, 2) * (1, 8)). I was using (1, 1) to check that we could reach the bridge itself. It's obviously better to use the actual user-provided values; done.

Reply:
done